src/Symfony/Component/Routing/Generator/Dumper/CompiledUrlGeneratorDumper.php

```diff
@@ -35,12 +37,69 @@ public function getCompiledRoutes(): array
                 $compiledRoute->getTokens(),
                 $compiledRoute->getHostTokens(),
                 $route->getSchemes(),
+                [],
             ];
         }
 
         return $compiledRoutes;
     }
 
+    public function getCompiledAliases(): array
+    {
+        $routes = $this->getRoutes();
+        $compiledAliases = [];
+        foreach ($routes->getAliases() as $name => $alias) {
+            $deprecations = [];
+
+            if ($alias->isDeprecated()) {
+                $deprecation = $alias->getDeprecation($name);
+
+                $deprecations[] = $deprecation;
+
+                trigger_deprecation($deprecation['package'], $deprecation['version'], $deprecation['message']);
+            }
+            $visited = [];
+            while ($routes->hasAlias($alias->getId())) {
+                $currentId = $alias->getId();
+                $searchKey = array_search($currentId, $visited);
+                $visited[] = $currentId;
+
+                if (false !== $searchKey) {
+                    throw new RouteCircularReferenceException($currentId, \array_slice($visited, $searchKey));
+                }
+
+                $alias = $routes->getAlias($currentId);
+
+                if ($alias->isDeprecated()) {
+                    $deprecation = $alias->getDeprecation($currentId);
+
+                    $deprecations[] = $deprecation;
+
+                    trigger_deprecation($deprecation['package'], $deprecation['version'], $deprecation['message']);
```

Comment:
targeting a deprecated alias should raise a deprecation, so this one is fine compared to the previous one